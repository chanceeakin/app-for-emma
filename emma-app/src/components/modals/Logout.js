// @flow
import React, { Component } from 'react'
import {
  Modal,
  Text,
  TouchableHighlight,
  TouchableOpacity,
  View,
  StyleSheet,
} from 'react-native'
import GradientWrapper from './../wrappers/GradientWrapper'
import { button, modal as modalStyle, pageLayout } from './../../styles'
import type { SettingsAction } from './../../actions/settings.js.flow'

const {
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  errorButton,
  errorText,
  backButton,
} = button

const { modal, modalTitle } = modalStyle
const { container } = pageLayout

const styles: StyleSheet = StyleSheet.create({
  container,
  modal,
  modalTitle,
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  errorButton,
  errorText,
  backButton,
})

type Props = {
  handleLogout: SettingsAction,
};

type State = {
  modalVisible: boolean,
};

export default class ModalComponent extends Component<Props, State> {
  state = {
    modalVisible: false,
  };

  setModalVisible(visible: boolean): void {
    this.setState({ modalVisible: visible })
  }

  render() {
    const { handleLogout } = this.props
    return (
      <View style={styles.container}>
        <Modal
          animationType="fade"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={() => {
            alert('Modal has been closed.')
          }}
        >
          <GradientWrapper color="orange" style={styles.modal}>
            <Text style={styles.modalTitle}>Logout</Text>
            <View>
              <TouchableOpacity onPress={handleLogout} style={styles.bigButton}>
                <Text style={styles.bigButtonText}>Logout</Text>
              </TouchableOpacity>
            </View>
            <TouchableHighlight
              style={styles.backButton}
              onPress={() => {
                this.setModalVisible(!this.state.modalVisible)
              }}
            >
              <Text>Back</Text>
            </TouchableHighlight>
          </GradientWrapper>
        </Modal>

        <TouchableOpacity
          style={styles.settingsButton}
          onPress={() => {
            this.setModalVisible(true)
          }}
        >
          <Text style={styles.settingsButtonText}>Logout</Text>
        </TouchableOpacity>
      </View>
    )
  }
}
