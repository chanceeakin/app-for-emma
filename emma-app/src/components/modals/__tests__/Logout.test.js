import React from 'react'
import Logout from '../Logout'

import renderer from 'react-test-renderer'

const props = {
  updatedEmail: 'email',
  updateEmailPatchTextField: jest.fn(),
  isPatchingEmailError: false,
  changeEmail: jest.fn(),
  isEmailModalShown: false,
  toggleEmailModal: jest.fn(),
}

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableOpacity')
})

jest.mock('TouchableHighlight', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

describe('<Logout />', () => {
  afterAll = () => {
    jest.mockRestore()
  }
  it('renders correctly', () => {
    const tree = renderer.create(<Logout {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
