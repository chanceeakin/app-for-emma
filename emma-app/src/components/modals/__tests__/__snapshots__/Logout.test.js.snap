// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <ExponentLinearGradient
      colors={
        Array [
          4294960596,
          4293904322,
        ]
      }
      endPoint={
        Array [
          1,
          1,
        ]
      }
      location={
        Array [
          0.25,
          0.4,
          1,
        ]
      }
      locations={null}
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingTop": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 45,
            "textAlign": "center",
          }
        }
      >
        Logout
      </Text>
      <View>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={undefined}
          style={
            Object {
              "borderColor": "#0D1B2A",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginTop": 20,
              "paddingHorizontal": 40,
              "paddingVertical": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#0D1B2A",
                "fontSize": 23,
                "fontWeight": "600",
              }
            }
          >
            Logout
          </Text>
        </TouchableOpacity>
      </View>
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[Function]}
        style={
          Object {
            "padding": 15,
          }
        }
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Back
        </Text>
      </TouchableHighlight>
    </ExponentLinearGradient>
  </Modal>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#D37482",
        "borderRadius": 5,
        "marginTop": 20,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fafafa",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Logout
    </Text>
  </TouchableOpacity>
</View>
`;
